config/config_test.go

```diff
@@ -92,3 +100,109 @@ func TestNewDefault(t *testing.T) {
 		t.Fatalf("config file not loaded")
 	}
 }
+
+// TestCreatePaths ensures that the paths are created when requested.
+func TestCreatePaths(t *testing.T) {
+	home, cleanup := Mktemp(t)
+	t.Cleanup(cleanup)
+
+	t.Setenv("XDG_CONFIG_HOME", home)
+
+	if err := config.CreatePaths(); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := os.Stat(config.Path()); err != nil {
+		if os.IsNotExist(err) {
+			t.Fatalf("config path '%v' not created", config.Path())
+		}
+		t.Fatal(err)
+	}
+
+	if _, err := os.Stat(filepath.Join(config.Path(), "repositories")); err != nil {
+		if os.IsNotExist(err) {
+			t.Fatalf("config path '%v' not created", config.Path())
+		}
+		t.Fatal(err)
+	}
+
+	// Trying to create when repositories path is invalid should error
+	_ = os.WriteFile("./invalidRepositoriesPath.txt", []byte{}, os.ModePerm)
```

Comment:
Won't this create garbage in working dir?

Reply:
Yes, which is the intention:  to create a garbage file which trggers an error when attempting to create the path.
It might be worth noting that this all takes place within a temp directory specifically to give the test a sandbox to play in 👍🏻